### ISSUE TITLE
Quoted arguments aren't parsed properly

### ISSUE THREAD

**Issue by Richard-Weiss** [modelcontextprotocol/inspector#96]
When entering quoted args like paths, they aren't split properly. The line `const origArgs = (query.args as string).split(/\s+/);` in `server/src/index.ts` splits arguments at whitespace, regardless of any quotes. Something like:
```
import { parse as shellParseArgs } from 'shell-quote';
const origArgs = shellParseArgs(query.args as string) as string[];
```
should be used instead.

**Steps to reproduce the behavior:**
1. Enter any any argument in quotes like a path in the arguments field
2. Hit connect
3. The MCP server will be unable to parse the falsely parsed args
![image](https://github.com/user-attachments/assets/37fc7b84-fe91-4948-ae6a-0bc79276c6a5)

**Expected behavior**
Quoted args should be respected and not split.

Here it is with that mitigation:
![image](https://github.com/user-attachments/assets/c6f3627f-e487-47cb-b3fa-b30119ba070c)

**Comment by ashwin-ant** [comment]
@Richard-Weiss thanks for filing! I think your proposed fix makes sense, feel free to open a PR 😄 Otherwise we may be able to get to this sometime next week.

**Comment by ashwin-ant** [comment]
Fixed in #104
